plugins/resources/src/mockarchive.cpp

```diff
@@ -184,19 +184,20 @@ irods::error make_hashed_path(
     }
     EVP_MD_free(message_digest);
 
-    char md5Buf[MAX_NAME_LEN]{};
-    std::strncpy(md5Buf, _path.c_str(), _path.size());
+    char buffer_to_hash[MAX_NAME_LEN]{};
+    std::strncpy(buffer_to_hash, _path.c_str(), _path.size());
 
     // Hash the specified buffer of bytes and store the results in the context.
-    if (0 == EVP_DigestUpdate(context, reinterpret_cast<unsigned char*>(md5Buf), _path.size())) {
+    if (0 == EVP_DigestUpdate(context, reinterpret_cast<unsigned char*>(buffer_to_hash), _path.size())) {
         const auto ssl_error_code = ERR_get_error();
         const auto msg = fmt::format("{}: Failed to calculate digest. error code: [{}]", __func__, ssl_error_code);
         THROW(DIGEST_UPDATE_FAILED, msg);
     }
 
     // Finally, retrieve the digest value from the context and place it into a buffer. Use the _ex function here
     // so that the digest context is not automatically cleaned up with EVP_MD_CTX_reset.
-    unsigned char hash[MAX_NAME_LEN];
+    static constexpr auto max_digest_length = EVP_MAX_MD_SIZE;
```

Comment:
so this length is now longer than MAX_NAME_LEN?

Reply:
Hmm... that's a good point. That 16 comes from the length of MD5 hashes, which is 128 bits, or 16 bytes. SHA256 hashes are 256 bits, or 32 bytes. We should probably update that (unless, of course, we don't really care that much about mockarchive's correctness on this front).

The new length is using `EVP_MAX_MD_SIZE` instead of 32 bytes because if we ever decide to move away from SHA256 in the future, we know that `EVP_DigestFinal` outputs *at most* `EVP_MAX_MD_SIZE` bytes, regardless of the hashing algorithm being used.

Reply:
seems good to keep them correct/consistent - for the next reader to not question everything again.

Reply:
Done